tests/kernel/fatal/message_capture/src/main.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2012-2014 Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+
+static int expected_reason = -1;
+
+void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
+{
+	printk("Caught system error -- reason %d\n", reason);
+
+	if (expected_reason == -1) {
+		printk("Was not expecting a crash\n");
+		k_fatal_halt(reason);
+	}
+
+	if (reason != expected_reason) {
+		printk("Wrong crash type got %d expected %d\n", reason,
+		       expected_reason);
+		k_fatal_halt(reason);
+	}
+
+	expected_reason = -1;
+}
+
+void main(void)
+{
+	unsigned int key;
+
+	expected_reason = K_ERR_KERNEL_OOPS;
+
+	key = irq_lock();
+	k_oops();
+	printk("SHOULD NEVER SEE THIS\n");
+	irq_unlock(key);
+}
```

Comment:
Please put the code in a function called test_*(), then call it in main().  And, add doxygen tags for that new test_*() fucntion like below.

/**
 * @brief Test preempt thread initialization via K_THREAD_DEFINE
 * @ingroup kernel_thread_tests
 *
 * @details
 * Test Objective:
 * -
 *
 * Prerequisite Conditions:
 * - N/A
 *
 * Input Specifications:
 * - N/A
 *
 * Test Procedure:
 * -# 
 * -# 
 * -# 
 * -# 
 * -# 
 * -# 
 *
 * Expected Test Result:
 * -
 *
 * Pass/Fail Criteria:
 * - Successful if
 * - Failure if
 *
 * Assumptions and Constraints:
 * - N/A
 */

Reply:
@yingmingx , please take look at this comment.